### DISCUSSION TITLE
如何给 ChatGLM3 增加知识储备

### DISCUSSION THREAD

**Discussion by ponponon** [zai-org/ChatGLM3#345] [Q&A] [answered]
比我有很多电影数据，包括电影的标题、别名、演员、年份、导演、剧情、简介、票房数据等等，还有对应电影的截图快照

我希望可以让 ChatGLM3 记住所有的这些信息，以后询问 ChatGLM3 的时候，我可以说出电影别名或者一段近似的简介或者一个画面截图，让它可以告诉我这是出自哪个电影

这样如何实现呢？

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
1. 外挂数据库，使用类似Langchain或者更顶层的框架，例如Langchain-Chatchat等来实现
2. 微调，但是你无法传入截图因为该模型不支持多模态

**Reply by ponponon** [reply]
「但是你无法传入截图因为该模型不支持多模态」—— https://www.chatglm.cn/main/detail 这个智谱清言支持图片输入，是怎么实现的呢？使用的不是 ChatGLM3 吗？

**Comment by zRzRzRzRzRzRzR** [comment]
不是GLM3

**Reply by ponponon** [reply]
这个智谱清言支持图片输入的这部分功能有开源吗？使用的还是哪个模型呀？是 CogVLM 吗？

**Reply by zRzRzRzRzRzRzR** [reply]
没有开源，你可以通过开源的glm3+ cogvlm自己搓一个，这个确实可以
